pkg/kube/controllers/extendedsecret/extendedsecret_reconciler.go

```diff
@@ -124,6 +127,7 @@ func (r *ReconcileExtendedSecret) Reconcile(request reconcile.Request) (reconcil
 func (r *ReconcileExtendedSecret) updateExSecret(ctx context.Context, instance *esv1.ExtendedSecret) error {
 	obj := instance.DeepCopy()
 	op, err := controllerutil.CreateOrUpdate(ctx, r.client, obj, func() error {
+		obj.Spec = instance.Spec
```

Comment:
Why is this change needed?

Reply:
It's helpful to update default `spec.request.signerType` if it's certificate.

Reply:
refactored CreateOrUpdate mutateFn